library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -409,7 +415,15 @@ public boolean isReady() {
    */
   @Override
   protected boolean hasOutputReady() {
-    return tunneling && buffersInCodecCount > 0 || super.hasOutputReady();
+    boolean fifoReady = true;
```

Comment:
Here we are trying to use FIFO length as renderer readiness criteria. If the frame rate is low such as 1fps on music channels the length would still be satisfactory and player wouldn't go into buffering state as long as there is a frame queued in the FIFO. If network bandwidth is low the FIFO length will get close to zero before decoder starts stalling. It's important to note that Broadcom's decoder will stall before rendering last few frames so zero FIFO length cannot be used as a criteria. The 333ms value is somewhat experimental. It was chosen to accommodate at least 10 frames of 30fps stream.

Reply:
@andrewlewis @ojw28  Where are you guys at with addressing this issue.  I'm preparing to update our code base to what will likely by your 2.11.5 release.   We would very much like to not keep merging conflicted changes to the `MediaCodecVideoRenderer`, for obvious reasons.

Pretty sure we can send you an STB (Broadcom based) that reproduces the issue #6366  I think I included the URL for a section of music choice content that reproduces the issue 100% of the time

We are also see this, as Dasha explained, in other streams where audio is buffered sufficiently ahead of video.  As I'm sure you guys are aware, Broadcom codecs often require some special spoon feeding.

Thanks again for looking at this with us, please tell me what I need to do to help the process along.